### ISSUE TITLE
redundant_field_names FP: path with type parameters

### ISSUE THREAD

**Issue by sinkuu** [rust-lang/rust-clippy#3476]
```rust
fn foo<T>() {
}

struct S {
    foo: fn(),
}

fn main() {
    S { foo: foo::<i32> };
}
```

```
warning: redundant field names in struct initialization
 --> src/main.rs:9:9
  |
9 |     S { foo: foo::<i32> };
  |         ^^^^^^^^^^^^^^^ help: replace it with: `foo`
  |
  = note: #[warn(clippy::redundant_field_names)] on by default
```

**Comment by oli-obk** [comment]
This *should* be fairly easy to fix.

Probaby enough to add checks to

https://github.com/rust-lang/rust-clippy/blob/f5831523d335bc57a6371806b2ea1978942ac490/clippy_lints/src/redundant_field_names.rs#L60

for the other fields of `path.segments[0]`.

The docs of the `PathSegment` type can be found in https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ast/struct.PathSegment.html
